sdk/core/core/inc/az_http_request_builder.h

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifndef AZ_HTTP_REQUEST_BUILDER_H
+#define AZ_HTTP_REQUEST_BUILDER_H
+
+#include <_az_cfg_prefix.h>
+
+#include <az_pair.h>
+#include <az_span.h>
+#include <az_write_span_iter.h>
+
+typedef struct {
+  az_span * buffer;
```

Comment:
The * should be removed and I'd love to see some comments on each line explaining the purpose of each field.
Commenting code is Critically Important because others MUST be able to read and maintain the code.
I'll mention commenting a lot.